### PR TITLE
Mongo toolkit issue#32 mongo match text

### LINKED ISSUE #32: Mongo_Toolkit: MongoMatchList won't' allow matching strings
Probably need a MongoMatchText one as well

### PR DESCRIPTION
**Pull request by emidio-piermarini**
### Issues addressed by this PR

Closes #32

### CODE REVIEW

File: Mongo_Engine/Create/MongoServer.cs [hunk 1]
```diff
@@ -53,15 +58,17 @@ public MongoServer(string folderName)
 
                 Thread.Sleep(1000);
                 m_Client = new MongoClient(@"mongodb://localhost:27017");
+                alert = "MongoDB Files Stored in " + folderName;
             }
+            return alert;
         }
```
Comment (epignatelli) [inline Mongo_Engine/Create/MongoServer.cs]: I'd reformat beside just porting. I am not sure why we kept the fields at this point. If you're calling the `mongod` command, your mongo is added to the path already, so when you start it you should be able to get the dbpath with something like `client.GetDatabase("admin").RunCommand<BsonDocument>("{getCmdLineOpts: 1}")["parsed"]["storage"]["dbPath"].ToString()`

File: Mongo_Test/Program.cs [hunk 2]
```diff
@@ -86,14 +86,14 @@ public static void Test()
...
             List<string> asdf3 = new List<string> {"M_X","F_X" };
-            string g = "";
-         //  g= BH.Engine.Mongo.Create.MongoCleanVariable(asdf, g);
+            List<string> g = new List<string>();
+            //  g= BH.Engine.Mongo.Create.MongoCleanVariable(asdf, g);
             Console.WriteLine(g);
             List<object> fffff = new List<object> { g};
             // g = BH.Engine.Mongo.Create.MongoGroup(asdf2,asdf3);
             //     Console.WriteLine(g);
-            g = BH.Engine.Mongo.Create.MongoMax("stocazzo");
```
[the remark this answers is not in the document]
Comment (emidio-piermarini) [reply Mongo_Test/Program.cs]: done

File: Mongo_Engine/Create/MongoMatchText.cs [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+    {
+        /***************************************************/
+        /**** Public Methods                            ****/
+        /***************************************************/
+        public static List<string> MongoMatchText(string Key, List<string> Filter)
+        {
+            string projectquery = "";
+            string matchquery = "";
+            string mongolist ="";
+            string tempVar = "";
```
[the remark this answers is not in the document]
Comment (emidio-piermarini) [reply Mongo_Engine/Create/MongoMatchText.cs]: deleted

**Comment by emidio-piermarini** [comment]
@epignatelli all comments on the Mongo Server should really be raised as separate issues and @adecler should be looped in for those.  Equally, in terms of cases for the methods, there's a lot that needs to be discussed going forward for the mongo queries and whats in here now is deffo just a first crack based on a project we had a year ago.

i'd be tempted to raise this as a separate issue as well.   We have another project starting up for which we will leverage mongo quite a bit and we could probably slowly start to adress these and other issues going foward to bring the MongoToolkit to the next level.

**Comment by emidio-piermarini** [comment]
@adecler  @epignatelli  can we have  chat on this before the weekly call this week?  I have some new code forthcoming and would like to start adding from the master so would be good to clean this one up and get it into the master before starting (these will be overlaps of new and old).  Let me know

**Comment by epignatelli** [comment]
As per our last chat @emidio-piermarini, https://github.com/BHoM/Mongo_Toolkit/pull/60 just got merged - feel free to attack this and add me as reviewer

**Comment by FraserGreenroyd** [comment]
During today's PR closure day, @IsakNaslundBh and myself went through a lot of old PRs, including this one.

It is our personal opinion that PRs such as this one do not serve a purpose by staying open, even when `parked`, due to their age, and their incompatibilities with the `master` branches. We feel it would be best for this PR (and others which we are commenting on today) to be closed without merge (but not necessarily delete the branches in case the original author wishes to try and make the changes again in the future).

@al-fisher @adecler what are your thoughts on this please?

**Comment by al-fisher** [comment]
Thanks both - agree with the above approach. Closing now seems sensible.
Being Mongo - will leave you @adecler to make final comment and close if happy. Thanks

**Comment by adecler** [comment]
I would keep this one open a little bit longer. The code from this PR doesn't exist anywhere else as far as I know and will probably be useful when I work on database queries this quarter. Should be able to close by the end of this quarter though.

**Comment by FraserGreenroyd** [comment]
> I would keep this one open a little bit longer. The code from this PR doesn't exist anywhere else as far as I know and will probably be useful when I work on database queries this quarter. Should be able to close by the end of this quarter though.

Any update on this @adecler as we're at the end of another quarter with this still open? :smile:

**Comment by FraserGreenroyd** [comment]
Discussed with @adecler during a call on ZeroCodeTools today and we have agreed that this PR can be closed as it is very out-of-date and unusable in its current state. This PR will exist as a record should we need the work in the future.

**Pull request closed without merging**